### PR TITLE
Done reorder between conda and pip

### LINKED ISSUE #6154: Update installation documentation
One suggestion made in #6051 was to update the installation section of the documentation to focus on anaconda for new users; see https://github.com/astropy/astropy/pull/6051#issuecomment-301335291 for the one idea for doing that. It was clear this topic needed more discussion, so was dropped in #6051.

The purpose of this issue is to remind to update the installation documentation at some point..

### PR DESCRIPTION
**Pull request by CosmicTiger**
### Description

This pull request is to address ...

EDIT: Fixes #6154

### CODE REVIEW

**Comment by pllim** [comment]
Hello. Can you please clarify if this PR is to resolve an existing issue?

**Comment by CosmicTiger** [comment]
Hello @pllim yes it is related with the issue https://github.com/astropy/astropy/issues/6154#issuecomment-703226116 in which @astrojuanlu mentioned the reorder between conda installation and pip that they supposed to be in that exact order as I mentioned. I did not touch anything of the default answer of the PR because it seems that is something by default of one of your files and I didn't want to messed up anything.

**Comment by bsipocz** [comment]
A lot has happened in the  past 3 years, so I'm not sure any more that suggesting conda first is still the desirable approach, even though the issue hasn't been triaged and closed (thanks @CosmicTiger to find it!)

cc-ing @mwcraig and @taldcroft as they were involved in the xrefed issue and wrote the original comment asking for this change.

**Comment by taldcroft** [comment]
> A lot has happened in the past 3 years, so I'm not sure any more that suggesting conda first is still the desirable approach.

I still think that conda is the simplest approach for typical users. In other words, I still agree with the statement:
```
If you are new to Python and/or do not have familiarity with `Python virtual
environments <https://docs.python.org/3/tutorial/venv.html>`_, then we recommend
starting by installing the `Anaconda Distribution
<https://www.anaconda.com/distribution/>`_. This works on all platforms (linux,
Mac, Windows) and installs a full-featured scientific Python in a user directory
without requiring root permissions.
```
The Anaconda powershell on Windows seems very useful to me when I need to do Windows testing / development. I also think conda makes it much easier to work with different versions of Python, and in particular for users on systems with an older Python to use the latest.

That said, this all reflects my own experience and I only use conda, so I might be missing developments that make everything just easily work on all platforms.

**Comment by mwcraig** [comment]
This reordering is fine. These days I lean more towards miniconda + custom environments but that is probably too much for a brand new user. Anaconda is still probably the best bet for those folks.

**Comment by pllim** [comment]
Re: Anaconda vs Miniconda

I'd rather people learn how to `conda create -n newenv python=X.Y` from the start than to use Anaconda and then complain things don't work when the `base` env gets dirty and they have no option but to re-install.

File: docs/install.rst
Comment (mwcraig) [inline docs/install.rst:142]: I think these two can now be included with the `conda install` command above. While there is a lag of a couple of hours between when the package is published on pypi and when it shows up in conda-forge it isn't much more than that these days.  Any advantage gained by installing these with `pip` is outweighed, I think, by the disadvantage of having to remember which things are upgraded by `pip` and which by `conda`.

**Review by mwcraig — CHANGES_REQUESTED** [review]
Just one small change (several lines to cut 😀 , one to modify) and this is ready to go!

**Comment by mwcraig** [comment]
> I'd rather people learn how to conda create -n newenv python=X.Y from the start than to use Anaconda and then complain things don't work when the base env gets dirty and they have no option but to re-install.

Agree! I teach new students this right away because it is so much easier to nuke an environment than your entire install. But that seems like a much bigger change than what @CosmicTiger had started.

**Comment by pllim** [comment]
True, it is a borderline scope creep. But if we're doing the update anyway... If @CosmicTiger does not wish to make changes outside of strict re-ordering, we can follow-up after this is merge.

**Review by astrofrog — COMMENTED** [review]
I'm tempted to say we keep pip first now as pip is a lot better than it used to be and conda is a lot worse :-/ But no strong opinions, so happy to go with what others think.

**Comment by astrojuanlu** [comment]
> I'm tempted to say we keep pip first now as pip is a lot better than it used to be and conda is a lot worse :-/ But no strong opinions, so happy to go with what others think.

This reflects my own views at the moment. But I don't know if this is because pip has objectively gotten better while conda has gotten worse, or because I have gained enough expertise to understand when should I use one or the other... I find myself having to explain PEP 518 & 517, wheels vs tarballs, and obscure `pip install` CLI arguments too often. In any case, pip has definitely improved in recent years, so perhaps it's a combination of both.

**Comment by taldcroft** [comment]
In terms of dependency resolution speed, my experience with conda is that it has gotten much better in most cases in the last couple of years. @astrofrog - are there problems you are aware of for the simple use cases of beginners?

I agree that miniconda is definitely the way to go for slightly experienced users.  So in the paragraph above I would say:
```
... starting by installing the `Anaconda Distribution <https://www.anaconda.com/distribution/>`_ or the `Miniconda installer <https://docs.conda.io/en/latest/miniconda.html>`_.
```

If / when pip 20.3 gets released including an actual dependency solver like conda, then we'll see how many problems come up. I predict a lot of people pinning `pip < 20.3`. :smile:

**Comment by taldcroft** [comment]
In another plug for conda, I note that we use it to manage and install Perl packages!

File: docs/install.rst
Comment (astrojuanlu) [reply docs/install.rst:142]: Hi @cosmictiger! Are you still working on this?

**Comment by astrojuanlu** [comment]
Pinging @CosmicTiger in case they want to address @mwcraig request, although I'm not sure if we have any conclusions for the debate we had here. If we decide this is an incremental improvement in the right direction, let's move forward, otherwise I'm not sure what would be the best way to draw the current consensus about the best way to install Astropy for newcomers.

**Comment by embray** [comment]
We haven't heard from the original PR author in a while, and I think based on backs-and-forth about conda, we are not in a hurry to make this change, right?

**Comment by taldcroft** [comment]
I would vote to just merge this as is, invoking the "perfect is the enemy of good" mantra. This PR accomplishes the simple goal of re-ordering the sections and it closes an issue!  We can come back to fixing the niggles later as needed.

Although there are outstanding differences of opinion on the conda vs. pip question, it is worth noting that numpy recommends conda for both beginning and advanced users (https://numpy.org/install/). For me it is a good thing to coalesce on a common recommendation.

**Comment by pllim** [comment]
Hi humans :wave: - this issue was labeled as **Close?** approximately 2 months ago. If you think this issue should not be closed, a maintainer should remove the **Close?** label - otherwise, I will close this issue in 7 days.

*If you believe I commented on this issue incorrectly, please report this [here](https://github.com/pllim/action-astropy-stalebot/issues)*

**Comment by pllim** [comment]
Ops... supposed to run for issues only. Please ignore, sorry.

**Pull request closed without merging**
